papermill/execute.py

```diff
@@ -41,24 +45,24 @@ def preprocess(self, nb, resources):
     # Reset the notebook.
     for cell in nb.cells:
         # Reset the cell execution counts.
-        if hasattr(cell, "execution_count"):
+        if cell.get("execution_count") is not None:
             cell.execution_count = None
 
         # Clear out the papermill metadata for each cell.
         cell.metadata['papermill'] = dict(
             exception=None,
             start_time=None,
             end_time=None,
-            duration=None
+            duration=None,
+            status=PENDING  # pending, running, completed
```

Comment:
We definitely need to start speccing the metadata more formally (JSON Schema). We can get this metadata acknowledged in commuter and nteract pretty easily as well. It would help if the metadata had a version too, for our own sake. It's also good to think about how we'll upconvert from old versions of the papermill metadata to new versions. In this case it's new fields, just something to keep in mind for stability and maintenance sanity.